File: netcore/Flights/Flights.Dal/Repositories/InMemoryFlightRepository.cs
using Flights.Core.Interfaces;
using Flights.Core.Model;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Flights.Dal.Repositories
{
    public class InMemoryFlightRepository : IFlightRepository
    {
        private readonly List<Flight> _flights = new List<Flight>();

        public InMemoryFlightRepository()
        {
            _flights.AddRange(
                new List<Flight>
                {
                    new Flight
                    {
                        Id =Guid.NewGuid().ToString(),
                        Airline = "BA",
                        Number = "123",
                        Origin = "DUB",
                        Destination = "LHR",
                        ScheduleTime = "2020.02.12 14:00:00",
                        Status = "On Time"
                    },
                    new Flight
                    {
                        Id = Guid.NewGuid().ToString(),
                        Airline = "BA",
                        Number = "456",
                        Origin = "DUB",
                        Destination = "LHR",
                        ScheduleTime = "2020.02.12 20:00:00",
                        Status = "On Time"
                    }
                }
            );
        }

        public Flight Get(string id)
        {
            return _flights.FirstOrDefault(f => f.Id == id);
        }

        public List<Flight> GetAll(string filter)
        {
            return _flights;
        }

        public void Insert(Flight flight)
        {
            flight.Id = Guid.NewGuid().ToString();
            _flights.Add(flight);
        }

        public void Update(Flight flight)
        {
            var match = _flights.FirstOrDefault(f => f.Id == flight.Id);
            if(match != null)
            {
                _flights.Remove(match);
                _flights.Add(flight);
            }
        }
        public void Delete(string id)
        {
            var match = _flights.FirstOrDefault(f => f.Id == id);
            if (match != null)
            {
                _flights.Remove(match);
            }
        }
    }
}

File: Java/src/main/java/flights/dal/InMemoryFlightRepository.java
package flights.dal;

import flights.core.interfaces.IFlightRepository;
import flights.core.model.Flight;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;
import org.springframework.stereotype.Component;

@Component
public class InMemoryFlightRepository implements IFlightRepository {

  private List<Flight> flights;

  public InMemoryFlightRepository(){
    flights = new ArrayList(Arrays.asList(
        new Flight(UUID.randomUUID().toString(), "2020.02.14 14:00:00", "123", "BA", "DUB", "LHR", "OnTime"),
        new Flight(UUID.randomUUID().toString(), "2020.02.14 14:30:00", "456", "BA", "DUB", "STN", "OnTime")));

  }

  public Flight get(String id) {
    List<Flight> matches =
        flights.stream()
        .filter(x -> x.getId().equals(id))
        .collect(Collectors.toList());

    if(matches != null && matches.size() > 0) {
      return matches.get(0);
    } else return null;
  }

  public List<Flight> getAll(String query) {
    return flights;
  }

  public Flight insert(Flight flight) {
    flight.setId(UUID.randomUUID().toString());
    flights.add(flight);
    return flight;
  }

  public Flight update(Flight flight) {
    delete(flight.getId());
    flights.add(flight);
    return flight;
  }

  public void delete(String id) {
    flights.removeIf(x -> x.getId().equals(id));
  }
}
